prov/efa/src/efa_mr.c

```diff
@@ -37,6 +37,11 @@
 static int efa_mr_reg(struct fid *fid, const void *buf, size_t len,
 		      uint64_t access, uint64_t offset, uint64_t requested_key,
 		      uint64_t flags, struct fid_mr **mr_fid, void *context);
+static int efa_mr_map_insert_shm_reg(struct efa_domain *efa_domain,
```

Comment:
My suggestion is:

1. efa_mr_cache_entry_reg() does not even call ibv_reg_mr().
2. merge efa_mr_map_insert_shm_reg() and efa_mr_core_reg() into efa_mr_reg
3. in efa_mr_cache_reg(), if (entry->use_cnt==1) call efa_mr_reg().

Reply:
I think the function name ```efa_mr_map_insert_shm_reg``` is too specific, and will need to be updated if we decide to do something more in the future (we might need to do something special for GPU). I would rename it ```efa_mr_alloc_other``` (or something like it) and leave a comment here  about what it does.